aws-codeguruprofiler-profilinggroup/src/main/java/software/amazon/codeguruprofiler/profilinggroup/AgentPermissionHelper.java

```diff
@@ -0,0 +1,46 @@
+package software.amazon.codeguruprofiler.profilinggroup;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import software.amazon.awssdk.services.codeguruprofiler.CodeGuruProfilerClient;
+import software.amazon.awssdk.services.codeguruprofiler.model.CodeGuruProfilerException;
+import software.amazon.awssdk.services.codeguruprofiler.model.GetPolicyRequest;
+import software.amazon.awssdk.services.codeguruprofiler.model.GetPolicyResponse;
+import software.amazon.cloudformation.proxy.AmazonWebServicesClientProxy;
+
+import java.util.List;
+import java.util.Map;
+
+import static java.util.Collections.emptyList;
+import static java.util.Collections.singletonList;
+
+public class AgentPermissionHelper {
+    private static final CodeGuruProfilerClient profilerClient = CodeGuruProfilerClientBuilder.create();
+    private static final ObjectMapper objectMapper = new ObjectMapper();
+
+    @FunctionalInterface
+    public interface GetPrincipalsFunction<Proxy, PgName, Principals> {
+        Principals apply(Proxy s, PgName t);
+    }
+
+    @SuppressWarnings("unchecked")
+    public static List<String> getPrincipalsFromPolicy(AmazonWebServicesClientProxy proxy, String pgName) {
+        GetPolicyRequest getPolicyRequest = GetPolicyRequest.builder().profilingGroupName(pgName).build();
+        GetPolicyResponse getPolicyResponse = proxy.injectCredentialsAndInvokeV2(getPolicyRequest, profilerClient::getPolicy);
+        String policyInJson = getPolicyResponse.policy();
+
+        if (policyInJson == null) return emptyList();
+
+        try {
+            Map<String, List<Map<String, Map<String, Object>>>> policyMap = objectMapper.readValue(policyInJson, Map.class);
```

Comment:
Minor: Perhaps include here a comment mentioning that the tests have an example policy (helps a lot in grokking what we're doing here)